tools/ports/__init__.py

```diff
@@ -71,6 +88,30 @@ def init_port(name, port):
   validate_port(port)
 
 
+def load_external_port(external_port):
+  need_clear_project_build = Ports.fetch_project(external_port.name, external_port.EXTERNAL_PORT, external_port.SHA512, True)
+  port_file = os.path.join(Ports.get_dir(), external_port.name, external_port.PORT_FILE)
+  module_name = f'tools.ports.external.{external_port.name}'
+  spec = importlib.util.spec_from_file_location(module_name, port_file)
+  local_port = importlib.util.module_from_spec(spec)
+  spec.loader.exec_module(local_port)
```

Comment:
Some of this code (these 3 lines at least) are shared with `load_port` .. perhaps we can factor out?

Reply:
I will factor it out, no problem.

Reply:
Completed